### PR TITLE
Add 'License' message to the annotation proto.

### PR DESCRIPTION
**Pull request by ensonic**
This information is used by the default html doc generator and so it is nice
to be able to provide it.

### CODE REVIEW

**Comment by ensonic** [comment]
Just realize this is only the first step, we track down, what needs to be done to make this appear in the final json too.

**Comment by ensonic** [comment]
Now it should be good to go.

**Comment by ensonic** [comment]
Hi, I only get the bazel build to work.
When I try to build with make, I get errors:
https://gist.github.com/ensonic/173b3cfb6032ee730320729d356f8534

File: protoc-gen-swagger/genswagger/template.go [hunk 1]
```diff
@@ -784,6 +784,17 @@ func applyTemplate(p param) (string, error) {
 					s.Info.Contact.Email = spb.Info.Contact.Email
 				}
 			}
+			if spb.Info.License != nil {
+				if s.Info.License == nil {
+					s.Info.License = &swaggerLicenseObject{}
```
Comment (achew22) [inline protoc-gen-swagger/genswagger/template.go]: I think you need to add the `License` field to the `Info` object. I'm surprised this isn't a compile time error

Comment (ensonic) [reply protoc-gen-swagger/genswagger/template.go]: Shouldn't that be generated from "message Info" define in openapiv2.proto? Also when I build it with bazel it actually works. I am getting the license fields in the generated json.

**Comment by ensonic** [comment]
Update when I run 'make test' localy I get:
```
go test -race github.com/grpc-ecosystem/grpc-gateway/...
warning: ignoring symlink /usr/local/google/home/ensonic/go/src/github.com/grpc-ecosystem/grpc-gateway/bazel-bin
warning: ignoring symlink /usr/local/google/home/ensonic/go/src/github.com/grpc-ecosystem/grpc-gateway/bazel-genfiles
warning: ignoring symlink /usr/local/google/home/ensonic/go/src/github.com/grpc-ecosystem/grpc-gateway/bazel-grpc-gateway
warning: ignoring symlink /usr/local/google/home/ensonic/go/src/github.com/grpc-ecosystem/grpc-gateway/bazel-out
warning: ignoring symlink /usr/local/google/home/ensonic/go/src/github.com/grpc-ecosystem/grpc-gateway/bazel-testlogs
# google.golang.org/grpc/grpclb/grpc_lb_v1/messages
../../../google.golang.org/grpc/grpclb/grpc_lb_v1/messages/messages.pb.go:59:30: undefined: proto.InternalMessageInfo
../../../google.golang.org/grpc/grpclb/grpc_lb_v1/messages/messages.pb.go:112:31: undefined: proto.InternalMessageInfo
../../../google.golang.org/grpc/grpclb/grpc_lb_v1/messages/messages.pb.go:160:40: undefined: proto.InternalMessageInfo
../../../google.golang.org/grpc/grpclb/grpc_lb_v1/messages/messages.pb.go:302:47: undefined: proto.InternalMessageInfo
../../../google.golang.org/grpc/grpclb/grpc_lb_v1/messages/messages.pb.go:358:33: undefined: proto.InternalMessageInfo
../../../google.golang.org/grpc/grpclb/grpc_lb_v1/messages/messages.pb.go:441:41: undefined: proto.InternalMessageInfo
../../../google.golang.org/grpc/grpclb/grpc_lb_v1/messages/messages.pb.go:590:48: undefined: proto.InternalMessageInfo
../../../google.golang.org/grpc/grpclb/grpc_lb_v1/messages/messages.pb.go:639:32: undefined: proto.InternalMessageInfo
../../../google.golang.org/grpc/grpclb/grpc_lb_v1/messages/messages.pb.go:698:28: undefined: proto.InternalMessageInfo
ok  	github.com/grpc-ecosystem/grpc-gateway/codegenerator	0.541s
# github.com/grpc-ecosystem/grpc-gateway/examples/clients/abe
examples/clients/abe/examplepb_a_bit_of_everything.go:47:13: undefined: ExamplepbNumericEnum
examples/clients/abe/examplepb_a_bit_of_everything.go:63:22: undefined: ExamplepbNumericEnum
examples/clients/abe/examplepb_a_bit_of_everything.go:73:22: undefined: ExamplepbNumericEnum
FAIL	github.com/grpc-ecosystem/grpc-gateway/examples/integration [build failed]
ok  	github.com/grpc-ecosystem/grpc-gateway/protoc-gen-grpc-gateway/descriptor	0.569s
ok  	github.com/grpc-ecosystem/grpc-gateway/protoc-gen-grpc-gateway/gengateway	0.590s
ok  	github.com/grpc-ecosystem/grpc-gateway/protoc-gen-grpc-gateway/httprule	0.556s
ok  	github.com/grpc-ecosystem/grpc-gateway/protoc-gen-swagger	0.549s
ok  	github.com/grpc-ecosystem/grpc-gateway/protoc-gen-swagger/genswagger	0.554s
FAIL	github.com/grpc-ecosystem/grpc-gateway/runtime [build failed]
ok  	github.com/grpc-ecosystem/grpc-gateway/utilities	0.549s
Makefile:169: recipe for target 'test' failed
make: *** [test] Error 2
``
Looks like https://github.com/google/protobuf/issues/4582. Will investigate.
In any case those are different than what travis reported and don't seem to be related to my change.

**Comment by ensonic** [comment]
Hi, I really need help to understand why the tests fail. What exactly should I run locally to build/run not using bazel. I have the git rooted in my gopath:
/home/ensonic/go/src/github.com/grpc-ecosystem/grpc-gateway

**Comment by achew22** [comment]
Did you try regenerating the `openapiv2.pb.go` file that is generated from the proto? It looks like we don't have an entry in the `Makefile` for it. Would you mind adding a new line in the [`$(SWAGGER_PLUGIN)`](https://github.com/grpc-ecosystem/grpc-gateway/blob/master/Makefile#L130) section of the makefile to invoke protoc like:

`protoc -I $(PROTOC_INC_PATH) -I. --plugin=$(GO_PLUGIN) --go_out=$(PKGMAP),plugins=grpc:$(OUTPUT_DIR) $(PROTOFILE)
`

**Comment by johanbrandhorst** [comment]
Hi @ensonic, we're made some changes to CI since you first had these problems, want to rebase and give it another go?

**Comment by johanbrandhorst** [comment]
Please rebase on master for new CI functionality.

**Comment by ensonic** [comment]
Done, lets see ...

**Comment by johanbrandhorst** [comment]
Interesting, is that because of a dependency update or something? Would you be interested in submitting a PR to regenerate the file, or, regenerate it in this PR?

**Comment by johanbrandhorst** [comment]
So if you look at our CI tests it's actually saying that you still need to regenerate things. Bazel doesn't use the generated files in the same way (apparently) but we still need them. The `License` type needs to be added to the generated file.

**Comment by johanbrandhorst** [comment]
Don't run just `make test` locally, please regenerate the files as per the contribution guidelines: https://github.com/grpc-ecosystem/grpc-gateway/blob/master/CONTRIBUTING.md#i-want-to-regenerate-the-files-after-making-changes

**Comment by ensonic** [comment]
This looks better. The part about regenerating files should be one of the commands in 'how-to-contribute' and should be done before sending a PR :)

**Comment by johanbrandhorst** [comment]
Nice one, thanks!

**Comment by johanbrandhorst** [comment]
Thanks for the contribution :). Our old CI system was largely to blame, unfortunately there's still lots of good contributions like this laying around.

**Pull request merged**